ipv8/dht/community.py

```diff
@@ -458,6 +463,18 @@ def on_find_response(self, peer, dist, payload):
             cache.deferred.callback((cache.node, {'values': payload.values} if payload.values
                                      else {'nodes': payload.nodes}))
 
+    def block_chunk_refresh(self):
+        """
+        Republish any TrustChain block chunks which are about to expire in the DHT
+        """
+        from ..REST.dht_endpoint import TC_KEY_SUFFIX
+        hashed_dht_key = hashlib.sha1(self.my_peer.public_key.key_to_bin() + TC_KEY_SUFFIX).digest()
+
+        # for value in self.storage.get(hashed_dht_key):
+        for value in self.storage.items[hashed_dht_key]:
```

Comment:
We don't know if the owner of the block stores its values in `Storage`, as it depends on the distance between node ID's. It's better to just query your TrustChain database to get the required values. Please only republish your latest block, to avoid ending up storing your entire TrustChain in the DHT.